hello_world_on_chain/lib/TAParserLib.sol

```diff
@@ -34,11 +46,11 @@ library TAParserLib {
         return address(uint160(uint256(keccak256(strippedPublicKey))));
     }
 
-    function parseTA(
-        string calldata taData,
-        address publicKeyAddress
+    function verifyTA(
```

Comment:
Biggest changes that I made to the structure of this is that the verify of the TA and the parsing of the data in the TA to FN call claims are decoupled.  That allows a user to more clearly see layers of the encoding and verification (and could allow us to easily add other transitively attested things)